tools/jit-times/Timestamp.cs

```diff
@@ -1,113 +1,53 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.Text.RegularExpressions;
 
 namespace jittimes {
-	public struct Timestamp : IComparable {
-		public Int64 seconds;
-		public int milliseconds;
-		public int nanoseconds;
-
+	public record struct Timestamp(long nanoseconds) : IComparable {
```

Comment:
While moving to store *only* nanoseconds within the Unix epoch is a massive simplification, it *also* reduces the amount of time that can be tracked.  As this is "nanoseconds of Unix epoch" in a signed 64-bit integer, the maximum time this approach can hold is around 2262-Apr-11.  Any date after 23h47min16sec on that date results in a negative value:

```csharp
(long) (new DateTime(2262, 4, 11, 23, 47, 17) - DateTime.UnixEpoch).TotalNanoseconds
// -9223372036854775808
```

Given that this is still 238 years away, this *probably* doesn't matter…

Reply:
No disrespect, but if MS folks think it makes sense write this comment to PR + make others read it... I suspect I understand why there is zero progress with some huge issues like [the one this PR is related](tohttps://github.com/dotnet/runtime/issues/101135).